go.mod

```diff
@@ -10,14 +10,14 @@ require (
 	github.com/open-policy-agent/cert-controller v0.3.0
 	github.com/prometheus/client_golang v1.12.1
 	go.uber.org/zap v1.21.0
-	k8s.io/api v0.23.4
-	k8s.io/apimachinery v0.23.4
-	k8s.io/client-go v0.23.4
-	k8s.io/component-base v0.23.3
+	k8s.io/api v0.24.0
```

Comment:
Do you have special reasons to update the dependences?

Reply:
I need PR https://github.com/kubernetes-sigs/controller-runtime/pull/1765 to fix delete validation error in ClusterQueue.

Reply:
@kerthcet Can we bump to k8s.io v1.24.0？

Reply:
I am concerned that this will introduce a major bump in the overall project. right now we have 1.22 in the docs as the minimum dependency. and users running v0.1 and v0.1.1 will expect this to be true at least for a couple more versions. so raising the bar to 1.24 before V0.2 is not a viable option IMO

Reply:
OK, maybe I can first remove the codes that depends on controller-runtime master and wait until upgrade later.